### DISCUSSION TITLE
Eliminate Classification Head in Mask R-CNN

### DISCUSSION THREAD

**Discussion by MountassirEl** [facebookresearch/detectron2#5030] [Q&A]
Hello everyone!

Is it possible to remove the classification head in Mask R-CNN. Currently, I have set the following configurations in my code:
```
# Whether to use class agnostic for bbox regression
cfg.MODEL.ROI_BOX_HEAD.CLS_AGNOSTIC_BBOX_REG = True
# Whether to use class agnostic for mask prediction
cfg.MODEL.ROI_MASK_HEAD.CLS_AGNOSTIC_MASK = True
```

Despite enabling class-agnostic settings for both bounding box regression and mask prediction, I am still observing classification loss during training. This seems to suggest that the classification head is still active, even though I have attempted to disable it.

I would appreciate any guidance on how to completely eliminate the classification head in Mask R-CNN. Alternatively, if there are any additional configurations or modifications I need to make, please let me know.

Thank you for your support.

**Comment by akjaouen** [comment]
I have a related question. I have been training a mask-RCNN model using detectron2 for the segmentation of a single type of objects (I only want to detect cars, and that's it). Since there is only one class, I don't understand why the classification loss evolves. Is there a way to fix it, so that I can focus only on the box and mask losses ? I understand this may be a broader topic than the detectron2 implementation. If you think so, could you point out another platform where I can ask this kind of question ?

Thanks !

[Screenshot 2023-07-24 at 17 30 21](https://github.com/facebookresearch/detectron2/assets/127221564/bc8310c2-224f-4e91-aee1-c8d0b07b2332)
